zap/src/test/java/org/parosproxy/paros/core/scanner/AlertUnitTest.java

```diff
@@ -157,4 +159,55 @@ void shouldBuildAlertWithNoTagsWhenOneRemoved() {
         // Then
         assertThat(tagCount, is(equalTo(0)));
     }
+
+    @ParameterizedTest
+    @ValueSource(ints = {-1, 0})
+    void shouldNotAddCweTagIfNotValidValue(int cweId) {
+        // Given
+        Builder builder = new Alert.Builder();
+        builder.setCweId(cweId);
+        Alert alert = builder.build();
+        // When
+        int tagCount = alert.getTags().size();
+        // Then
+        assertThat(tagCount, is(equalTo(0)));
+    }
+
+    @Test
+    void shouldAddCweTagWhenPlausibleValue() {
+        // Given
+        int cwe = 618;
+        String cweUrl = "https://cwe.mitre.org/data/definitions/618.html";
+        String cweKey = "CWE-" + cwe;
+        Builder builder = new Alert.Builder();
+        builder.setCweId(cwe);
+        Alert alert = builder.build();
+        // When
+        int tagCount = alert.getTags().size();
+        // Then
+        Map<String, String> tags = alert.getTags();
+        assertThat(tagCount, is(equalTo(1)));
+        assertThat(tags.containsKey(cweKey), is(equalTo(true)));
+        assertThat(tags.get(cweKey), is(equalTo(cweUrl)));
+    }
+
+    @Test
+    void shouldAddOneCweTagEvenIfCweIdSetTwice() {
+        // Given
+        int cwe = 618;
+        int cwe2 = 619;
+        String cweUrl = "https://cwe.mitre.org/data/definitions/619.html";
+        String cweKey = "CWE-" + cwe2;
+        Builder builder = new Alert.Builder();
+        builder.setCweId(cwe);
+        builder.setCweId(cwe2);
```

Comment:
Call build before this line as well (the tag needs to be added to the alert rather than the builder).

Reply:
The semantics of the (new) test were changed, the second call to `setCweId` should still be done to the builder.

Reply:
Ok, I misinterpreted something said earlier. Will tweak.